src/operator/batch_norm-inl.h

```diff
@@ -50,20 +52,294 @@ struct BatchNormParam : public dmlc::Parameter<BatchNormParam> {
   }
 };
 
-template<typename xpu>
-class BatchNormOp : public Operator {
+/*! \brief Batch normalization operator */
+template<typename xpu, typename Dtype, typename AccType>
+class BatchNormOp : public Operator
+                  , public Callbacker<Operator> {
```

Comment:
It is convenient because I can use dynamic_cast in the unit tests to check for existence of the interface.  Otherwise, for other operators, a change would have to be made to add a function returning a pointer to the callback interface, etc.